pkg/utils/feed.go

```diff
@@ -136,17 +137,19 @@ func FindObsoletedFeeds(oldFeeds []appsapi.Feed, newFeeds []appsapi.Feed) []apps
 	return obsoleteFeeds
 }
 
-func FindBasesFromUIDs(baseLister applisters.BaseLister, uids []string) []*appsapi.Base {
+func FindBasesFromUIDs(baseIndexer cache.Indexer, uids []string) []*appsapi.Base {
 	allBases := []*appsapi.Base{}
 	for _, uid := range uids {
-		bases, err := baseLister.List(labels.SelectorFromSet(labels.Set{
-			uid: "Base",
-		}))
-		if err != nil {
+		if objs, err := baseIndexer.ByIndex("uid", uid); err != nil {
 			klog.ErrorDepth(5, err)
 			continue
+		} else {
+			if len(objs) == 1 {
+				allBases = append(allBases, objs[0].(*appsapi.Base))
```

Comment:
The length of `objs` could only be 0 or 1. Right?

Reply:
yes, find by base uid so result is unique.